accepted/0000-caasp-nodes.md

```diff
@@ -0,0 +1,274 @@
+
+The risk of breaking the cluster by issuing any forbidden operation is still present while the blackout is disabled.
+
+Value of this solution:
+* Easy and quick to implement
+* The user can still break its cluster by issuing any forbidden action while the cluster is not in the blackout
+* The Workflow for the user is different and mechanical, can lead to errors
+
+#### Uyuni/SUSE Manager action locking
+
+The idea is that Uyuni/SUSE Manager must restrict the following features:
+
+- Reboot a system:
+  - If the targeted minion is a `caasp_node` system type, the action for rebooting a system:
+    - Must be hidden from the UI and the corresponding action must fail: code paths `/systems/details/RebootSystem`, `/systems/ssm/misc/RebootSystem`, `ActionChainManager#scheduleRebootAction`, `ActionChainManager#scheduleRebootActions`
+    - The corresponding action must fail if scheduled via XMLRPC: `SystemHandler#scheduleReboot`
+- Package upgrade:
+  - If the targeted minion is a `caasp_node` system type, the action for upgrade a package:
+    - Must be hidden from the UI and the corresponding action must fail: code paths `/systems/details/packages/UpgradableList`, `systems/ssm/PackageUpgrade`, `ActionChainManager#schedulePackageUpgrade`, `ActionChainManager#schedulePackageUpgrades`
+    - The corresponding action must fail if scheduled via XMLRPC: [`SystemHandler#schedulePackageInstall`]
+- Package removal:
+  - If the targeted minion is a `caasp_node` system type, the action for removing a package:
+    - Must be hidden from the UI and the corresponding action must fail: `/systems/details/packages/RemoveConfirm`, `systems/ssm/PackageRemove`, `ActionChainManager#schedulePackageRemoval`, `ActionChainManager#schedulePackageRemoval`
+    - The corresponding action must fail if scheduled via XMLRPC: `SystemHandler#schedulePackageRemove`
+- Patch apply:
+  - If the targeted minion is a `caasp_node` system type, the action for applying a patch:
+    - Must be hidden from the UI and the corresponding action must fail: `/systems/details/ErrataList.do`, `/systems/ssm/ListErrata`, `ErrataManager#applyErrata`
+    - The corresponding action must fail if scheduled via XMLRPC: [`SystemHandler#scheduleApplyErrata`]
+
+Value of this solution:
+* This workflow does not constitute a radical change for a user
+* Change does not involve Salt, the user can still break its cluster by issuing Salt commands
+* Requires changes in Uyuni/SUSE Manager core code
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+<!-- Why should we **not** do this?
+
+  * obscure corner cases
+  * will it impact performance?
+  * what other parts of the product will be affected?
+  * will the solution be hard to maintain in the future? -->
+
+The user can craft a Salt state that executes one of the forbidden operations and breaks the cluster. This possibility is outside the scope of the RFC.
+
+# Alternatives
+[alternatives]: #alternatives
```

Comment:
Could enhancing the minion blackout feature in Salt, so that individual states can be whitelisted, be another option?

Reply:
Yes, it might work indeed. Adding it to the alternatives.